### PR TITLE
migrated to androidx

### PR DESCRIPTION
**Pull request by kibotu**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 4.0.0 (21.8.2018)
```
Comment (ZacSweers) [inline CHANGELOG.md:3]: Leave this out for now, maintainers will update the changelog when a new release is made

File: CalligraphySample/build.gradle [hunk 2]
```diff
@@ -20,17 +21,22 @@ android {
...
 }
 
 dependencies {
-    compile project(':calligraphy')
-    compile 'io.github.inflationx:viewpump:1.0.0'
-    compile 'com.android.support:support-v4:24.0.0'
-    compile 'com.android.support:appcompat-v7:24.0.0'
+    implementation project(':calligraphy')
+    // implementation 'io.github.inflationx:viewpump:1.0.0'
+    implementation 'com.github.kibotu:ViewPump:v2.0.0'
```
Comment (ZacSweers) [inline CalligraphySample/build.gradle:34]: What's this?

File: CalligraphySample/src/main/java/io/github/inflationx/calligraphy3/sample/MainActivity.java [hunk 3]
```diff
@@ -37,14 +36,16 @@ public void run() {
         }, 1000);
 
         handler.postDelayed(new Runnable() {
-            @Override public void run() {
```
Comment (ZacSweers) [inline CalligraphySample/src/main/java/io/github/inflationx/calligraphy3/sample/MainActivity.java:40]: Nit - don't change the style here

File: CalligraphySample/src/main/java/io/github/inflationx/calligraphy3/sample/PlaceholderFragment.java [hunk 4]
```diff
@@ -23,19 +25,19 @@ public static Fragment getInstance() {
     }
 
     @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle state) {
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle state) {
```
Comment (ZacSweers) [inline CalligraphySample/src/main/java/io/github/inflationx/calligraphy3/sample/PlaceholderFragment.java:28]: This project assumes nonnull is the default, please only use nullable annotations as needed

File: CalligraphySample/src/main/java/io/github/inflationx/calligraphy3/sample/TextField.java [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
-import android.widget.TextView;
+
+import androidx.appcompat.widget.AppCompatTextView;
 
 /**
  * Created by chris on 17/03/15.
  * For Calligraphy.
  */
-public class TextField extends TextView {
+public class TextField extends AppCompatTextView {
```
Comment (ZacSweers) [inline CalligraphySample/src/main/java/io/github/inflationx/calligraphy3/sample/TextField.java:13]: What's this change for?

File: CalligraphySample/build.gradle [hunk 6]
```diff
@@ -20,17 +21,22 @@ android {
...
+    implementation 'androidx.appcompat:appcompat:1.0.0-rc02'
 
-    compile 'com.jakewharton:butterknife:8.2.1'
+    implementation 'com.jakewharton:butterknife:9.0.0-SNAPSHOT'
+    annotationProcessor 'com.jakewharton:butterknife-compiler:9.0.0-SNAPSHOT'
 
-//    debugCompile 'com.squareup.leakcanary:leakcanary-android:1.3.1'
-    debugCompile 'com.squareup.leakcanary:leakcanary-android-no-op:1.3.1'
-    releaseCompile 'com.squareup.leakcanary:leakcanary-android-no-op:1.3.1'
+    debugImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.4'
```
Comment (ZacSweers) [inline CalligraphySample/build.gradle:40]: Latest is 1.6.0

File: README.md [hunk 7]
```diff
@@ -1,4 +1,4 @@
-Calligraphy
+Calligraphy [![API](https://img.shields.io/badge/API-15%2B-brightgreen.svg?style=flat)](https://android-arsenal.com/api?level=14) [![Gradle Version](https://img.shields.io/badge/gradle-4.10.2-green.svg)](https://docs.gradle.org/current/release-notes) [![androidx](https://img.shields.io/badge/androidx-1.0.0--rc02-brightgreen.svg)](https://developer.android.com/topic/libraries/support-library/refactor)
```
Comment (ZacSweers) [inline README.md:1]: Let's keep this out

File: build.gradle [hunk 8]
```diff
@@ -1,28 +1,26 @@
-// Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
   repositories {
+    google()
     jcenter()
   }
   dependencies {
-    classpath 'com.android.tools.build:gradle:2.2.1'
+    classpath 'com.android.tools.build:gradle:3.3.0-alpha11'
+    classpath 'com.github.dcendents:android-maven-gradle-plugin:2.1'
```
Comment (ZacSweers) [inline build.gradle:8]: What's this?

File: build.gradle [hunk 9]
```diff
@@ -1,28 +1,26 @@
-// Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
   repositories {
+    google()
     jcenter()
   }
   dependencies {
-    classpath 'com.android.tools.build:gradle:2.2.1'
+    classpath 'com.android.tools.build:gradle:3.3.0-alpha11'
```
Comment (ZacSweers) [inline build.gradle:7]: Please use the latest stable

File: build.gradle [hunk 10]
```diff
@@ -1,28 +1,26 @@
...
   // Is Release Build?
-  version = getProperty('VERSION_NAME');
+  version = getProperty('VERSION_NAME')
   ext {
     isReleaseVersion = has("release")
     versionCodeInt = getProperty('VERSION_CODE').toInteger()
   }
-}
-
-task wrapper(type: Wrapper) {
```
Comment (ZacSweers) [inline build.gradle:25]: Why remove this vs update it?

File: build.gradle [hunk 11]
```diff
@@ -1,28 +1,26 @@
...
   dependencies {
-    classpath 'com.android.tools.build:gradle:2.2.1'
+    classpath 'com.android.tools.build:gradle:3.3.0-alpha11'
+    classpath 'com.github.dcendents:android-maven-gradle-plugin:2.1'
   }
 }
 
 allprojects {
   repositories {
+    maven { url 'https://jitpack.io' }
```
Comment (ZacSweers) [inline build.gradle:14]: What's this for?

File: calligraphy/build.gradle [hunk 12]
```diff
@@ -18,11 +19,20 @@ android {
...
+    compileOnly 'androidx.appcompat:appcompat:1.0.0-rc02'
+    // implementation 'io.github.inflationx:viewpump:1.0.0'
+    implementation 'com.github.kibotu:ViewPump:v2.0.0'
+
+    testImplementation 'androidx.annotation:annotation:1.0.0-rc02'
+    testImplementation 'androidx.test:runner:1.1.0-alpha4'
 }
 
-apply from: rootProject.file('gradle/deploy.gradle')
+//apply from: rootProject.file('gradle/deploy.gradle')
```
Comment (ZacSweers) [inline calligraphy/build.gradle:38]: Why is this commented?

**Review by ZacSweers — COMMENTED** [review]
Thanks for the PR! Looks like there's a few things going on here, so let's try to keep it a pure migration and keep functional changes to a separate discussion. Also please abide by the existing code style (formatting, nullability annotations)

**Review by jbarr21 — CHANGES_REQUESTED** [review]
please address remaining feedback

**Comment by jbarr21** [comment]
Thanks for your contribution! I've made the changes for the comments that were requested and have merged it manually into master.

**Pull request closed without merging**